File: DayCodeBase/Day1.cs
using System.Collections.Generic;
using System.Linq;

namespace AdventOfCode2018.DayCodeBase
{
	public class Day1: DayCodeBase
	{
		public override string Problem1()
		{
			return GetData()
				.Select(int.Parse)
				.Sum()
				.ToString();
		}

		public override string Problem2()
		{
			var currentFrequency = 0;
			var seenFrequencies = new HashSet<int>(new[] { currentFrequency });
			var numbers = GetData().Select(int.Parse).ToList();
			while (true)
			{
				foreach (var number in numbers)
				{
					currentFrequency += number;
					if (seenFrequencies.Contains(currentFrequency))
						return currentFrequency.ToString();
					seenFrequencies.Add(currentFrequency);
				}
			}
		}
	}
}

File: aocFSharp/DayCodeBase/Day1.fs
module Day1

open System.Collections.Generic

let private data = DayCode.getData ("Day1_0.txt")

let private numbers = data |> Seq.map(int)

let private sum = fun acc elem -> acc + elem

let Program1 () = numbers |> Seq.reduce(sum) |> string

let Program2 () = 
    let mutable currentFrequency = 0
    let seenFrequencies = new HashSet<int>()

    while seenFrequencies.Contains currentFrequency |> not do
        for number in numbers do
            if (seenFrequencies.Contains currentFrequency |> not) then
                seenFrequencies.Add currentFrequency |> ignore
                currentFrequency <- currentFrequency + number
    currentFrequency |> string
